pkg/discovery/processor/cluster_processor.go

```diff
@@ -170,7 +170,8 @@ func (p *ClusterProcessor) DiscoverCluster() (*repository.ClusterSummary, error)
 	}
 	glog.V(2).Infof("Discovering cluster with %d nodes and %d pods.", len(nodeList), len(podList))
 	// Create kubeCluster and compute cluster resource
-	kubeCluster := repository.NewKubeCluster(svcID, nodeList).WithPods(podList)
+	kubeCluster := repository.NewKubeCluster(svcID, nodeList).WithPods(podList).
+		WithMachineSetToNodeUIDsMap(p.clusterInfoScraper.GetMachineSetToNodeUIDsMap(nodeList))
```

Comment:
Not required for this review, this is a general comment on the code structure. Since there are multiple ways of discovering node pools, it would be better to create a NodePoolProcessor in line with the NamespaceProcessor, ControllerProcessor etc and update the KubeCluster with all the node pools in one map.